lib/src/internal/shard/shard_manager.dart

```diff
@@ -119,14 +126,32 @@ class ShardManager implements IShardManager {
 
   /// Starts shard manager
   ShardManager(this.connectionManager, this.maxConcurrency) {
-    numShards = connectionManager.client.options.shardCount != null ? connectionManager.client.options.shardCount! : connectionManager.recommendedShardsNum;
+    totalNumShards = connectionManager.client.options.shardCount ?? connectionManager.recommendedShardsNum;
+    numShards = connectionManager.client.options.shards?.length ?? totalNumShards;
 
-    if (numShards < 1) {
+    if (totalNumShards < 1) {
       throw UnrecoverableNyxxError("Number of shards cannot be lower than 1.");
     }
 
+    List<int> toSpawn;
+    if (connectionManager.client.options.shards != null) {
+      if (connectionManager.client.options.shardCount == null) {
+        throw UnrecoverableNyxxError('Cannot specify shards to spawn without specifying total number of shards');
+      }
+      // Clone list to prevent original list from being modified with removeLast()
+      toSpawn = List.of(connectionManager.client.options.shards!);
```

Comment:
is that `List.of` needed? I think `connectionManager.client.options.shards` alone should work

Reply:
Read the comment above

Reply:
The comment above the line explains why it is needed

Reply:
ah sorry, Im scatterbrained sometimes :)